Library/Styles/Colors/AdaptiveColors.swift

```
@@ -0,0 +1,33 @@
+import UIKit
+
+/// A protocol for handling adaptive colors using a `RawRepresentable` enum.
+/// It dynamically constructs the color name based on its namespace and retrieves it from the asset catalog.
+protocol AdaptiveColors: RawRepresentable where RawValue == String {
+  /// Retrieves the adaptive color from the asset catalog.
+  /// - Returns: A `UIColor` object from the asset catalog, or default white if not found.
+  func adaptive() -> UIColor
+}
+
+extension AdaptiveColors {
+  public func adaptive() -> UIColor {
+    // Determines the appropriate namespace based on the enum type.
+    let namespace: String
+
+    switch Self.self {
+    case is Colors.Background.Type:
+      namespace = "background/"
+    case is Colors.Border.Type:
+      namespace = "border/"
+    case is Colors.Text.Type:
+      namespace = "text/"
+    default:
+      namespace = ""
```

Based on this PR it looks like all the colors are defined in those three prefixes, which would suggest that we wouldn't expect the default case to be hit. Is that correct? If this should NOT be hit and we cant count on the compiler catching it, can we add an assertion failure here?

---

That's a good point! However, I think we should keep the default case without an `assertionFailure` because we might want to define new color categories outside of a namespace.

---

If we do that in the future, someone adding those colors would hit the assertion and they could remove it. In the current state of the code it looks like a mistake to add it here, so IMO it should get treated that way.

---

That's a good point. I see your reasoning here, and it makes sense to catch unexpected cases early rather than silently defaulting to an empty namespace. I'll go ahead and add the assertionFailure to make it more explicit when a new color category is introduced outside of the expected namespaces.